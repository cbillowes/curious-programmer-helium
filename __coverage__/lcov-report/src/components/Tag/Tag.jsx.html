<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Tag/Tag.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Tag</a> Tag.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">114x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">114x</span>
<span class="cline-any cline-yes">114x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from "react"
import Link from "gatsby-link"
const _ = require("lodash")
&nbsp;
class Tag extends Component {
  render() {
    const { to } = this.props
&nbsp;
    <span class="missing-if-branch" title="else path not taken" >E</span>if (to) {
      return (
        &lt;Link key={`tag-${_.uniqueId()}-${to}`} className="tag" to={to}&gt;
          {this.props.children}
        &lt;/Link&gt;
      )
    } else {
<span class="cstat-no" title="statement not covered" >      return (</span>
        &lt;span key={`tag-${_.uniqueId()}-${_.random(9999)}`} className="tag"&gt;{this.props.children}&lt;/span&gt;
      )
    }
  }
}
&nbsp;
export default Tag</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 14:17:11 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
