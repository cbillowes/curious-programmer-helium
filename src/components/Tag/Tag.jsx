import React, { Component } from "react"
import Link from "gatsby-link"
const _ = require("lodash")

class Tag extends Component {
  render() {
    const { to } = this.props

    if (to) {
      return (
        <Link key={`tag-${_.uniqueId()}-${to}`} className="tag" to={to}>
          {this.props.children}
        </Link>
      )
    } else {
      return (
        <span key={`tag-${_.uniqueId()}-${_.random(9999)}`} className="tag">{this.props.children}</span>
      )
    }
  }
}

export default Tag