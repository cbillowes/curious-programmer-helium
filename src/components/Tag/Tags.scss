@import "../../layout/variables.scss";

.tags {
    margin: 0;
    padding: 0;

    li {
        list-style: none;
        display: inline;
    }

    .tag,
    &:visited {
        font-family: $font-family-sans-serif;
        line-height: 2em;
        font-size: 0.8em;
        background-color: $background-lighter;
        border-radius: $border-radius;
        color: $foreground;
        margin: 0.25em;
        padding: 0 1em;
        display: inline-block;
        text-shadow: none;
        font-weight: $font-family-sans-serif-heavy;

        &:hover {
            background-color: $primary;
            color: $background-lighter;
        }
    }
}
