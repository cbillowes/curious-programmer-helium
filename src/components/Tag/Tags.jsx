import React, { Component } from "react"
import Tag from "./Tag"
const _ = require("lodash")
import "./Tags.scss"

class Tags extends Component {
  render() {
    const tags = this.props.tags ? this.props.tags : []

    return (
      <ul className="tags">
        {tags.map((tag) => {
          return <li key={`tags-${_.kebabCase(tag)}`}>
            <Tag to={`/tags/${_.kebabCase(tag)}`}>{tag}</Tag>
          </li>
        })}
      </ul>
    )
  }
}

export default Tags