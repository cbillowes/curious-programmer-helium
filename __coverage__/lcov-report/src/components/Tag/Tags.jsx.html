<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Tag/Tags.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Tag</a> Tags.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">58x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">58x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">114x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from "react"
import Tag from "./Tag"
const _ = require("lodash")
import "./Tags.scss"
&nbsp;
class Tags extends Component {
  render() {
    const tags = this.props.tags ? this.props.tags : []
&nbsp;
    return (
      &lt;ul className="tags"&gt;
        {tags.map(tag =&gt; {
          return &lt;li key={`tags-${_.kebabCase(tag)}`}&gt;
            &lt;Tag to={`/tags/${_.kebabCase(tag)}`}&gt;{tag}&lt;/Tag&gt;
          &lt;/li&gt;
        })}
      &lt;/ul&gt;
    )
  }
}
&nbsp;
export default Tags</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 14:17:11 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
