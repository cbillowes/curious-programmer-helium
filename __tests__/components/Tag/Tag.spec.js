import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Tag from "../../../src/components/Tag/Tag"

describe("a tag component", () => {
  it("should render", () => {
    const component = <Tag />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render contents", () => {
    const component = mount(<Tag to="/privacy-policy">Privacy Policy</Tag>)
    expect(component.html()).toEqual(`<a class="tag" href="/privacy-policy">Privacy Policy</a>`)
  })

  it("should render linkless", () => {
    const component = mount(<Tag>Privacy Policy</Tag>)
    expect(component.html()).toEqual(`<span class="tag">Privacy Policy</span>`)
  })
})