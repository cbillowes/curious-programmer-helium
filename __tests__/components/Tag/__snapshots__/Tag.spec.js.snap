// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a tag component should render 1`] = `
<span
  className="tag"
/>
`;
